apps/np-atobarai/src/app/api/webhooks/saleor/transaction-refund-requested/refund-strategy/after-fulfillment-strategies.ts

```diff
@@ -0,0 +1,377 @@
+import { err, ok, Result } from "neverthrow";
+
+import { createLogger } from "@/lib/logger";
+import { createAtobaraiCancelTransactionPayload } from "@/modules/atobarai/api/atobarai-cancel-transaction-payload";
+import {
+  AtobaraiFulfillmentReportPayload,
+  createAtobaraiFulfillmentReportPayload,
+} from "@/modules/atobarai/api/atobarai-fulfillment-report-payload";
+import {
+  AtobaraiRegisterTransactionPayload,
+  createAtobaraiRegisterTransactionPayload,
+} from "@/modules/atobarai/api/atobarai-register-transaction-payload";
+import { CreditCheckResult } from "@/modules/atobarai/api/atobarai-transaction-success-response";
+import { IAtobaraiApiClient } from "@/modules/atobarai/api/types";
+import { createAtobaraiCustomer } from "@/modules/atobarai/atobarai-customer";
+import { createAtobaraiDeliveryDestination } from "@/modules/atobarai/atobarai-delivery-destination";
+import {
+  PartialRefundWithLineItemsGoodsBuilder,
+  PartialRefundWithoutLineItemsGoodsBuilder,
+} from "@/modules/atobarai/atobarai-goods/refund-goods-builders";
+import { createAtobaraiMoney } from "@/modules/atobarai/atobarai-money";
+import { createAtobaraiShopOrderDate } from "@/modules/atobarai/atobarai-shop-order-date";
+import {
+  AtobaraiTransactionId,
+  createAtobaraiTransactionId,
+} from "@/modules/atobarai/atobarai-transaction-id";
+import { createSaleorTransactionToken } from "@/modules/saleor/saleor-transaction-token";
+import {
+  RefundFailureResult,
+  RefundSuccessResult,
+} from "@/modules/transaction-result/refund-result";
+
+import { MalformedRequestResponse } from "../../saleor-webhook-responses";
+import { TransactionRefundRequestedUseCaseResponse } from "../use-case-response";
+import { AfterFulfillmentRefundContext, RefundStrategy } from "./types";
+
+export class AfterFulfillmentFullRefundStrategy implements RefundStrategy {
+  private readonly logger = createLogger("AfterFulfillmentFullRefundStrategy");
+
+  async execute(
+    context: AfterFulfillmentRefundContext,
+  ): Promise<Result<TransactionRefundRequestedUseCaseResponse, MalformedRequestResponse>> {
+    const { atobaraiTransactionId, apiClient } = context;
+
+    const payload = createAtobaraiCancelTransactionPayload({
+      atobaraiTransactionId,
+    });
+
+    const cancelResult = await apiClient.cancelTransaction(payload, {
+      checkForMultipleResults: true,
```

Comment:
maybe this flag will be better if it's explicitly "rejectMultipleResults". Because this is what matters - rejecting, not checking